ribbon/contract.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#----------------------------------------------------------------------------
+# Created By: Steven (steven@ribbon.finance)
+# Created Date: 04/04/2022
+# version ='0.01'
+# ---------------------------------------------------------------------------
+''' Abstract class for contract factory '''
+# ---------------------------------------------------------------------------
+
+# ---------------------------------------------------------------------------
+# Imports
+# ---------------------------------------------------------------------------
+from web3 import Web3
+import json
+
+# ---------------------------------------------------------------------------
+# Contract Factory
+# ---------------------------------------------------------------------------
+class ContractFactory:
```

Comment:
We can avoid `Factory` to name of classes in python,
as that's what they naturally allow you to do, create objects.

Reply:
We may want to instead call this `ContractConnection` ?

Reply:
Changed to `ContractConnection`

Reply:
Changed to `ContractConnection`